src/App.tsx

```diff
@@ -1,98 +1,77 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { currentItems, numberOfPages } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const itemsNumber = ['3', '5', '10', '20'];
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState<string>(itemsNumber[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const items = currentItems(42, currentPage, itemsPerPage);
+
+  const handleOption = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setItemsPerPage(value);
+    setCurrentPage(1);
+  };
+
+  const onPageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const prevButton = () => {
+    setCurrentPage(current => Math.max(1, current - 1));
+  };
+
+  const nextButton = () => {
+    setCurrentPage(current => Math.min(
+      numberOfPages(items.length, itemsPerPage), current + 1,
+    ));
+  };
+
+  const currentInfo = `Page ${currentPage} (items ${items[0]} - ${items.slice(-1)} of ${items.length})`;
```

Comment:
Oh, it seems like with your new function `currentItems` (I'd name it `getCurrentItems`) there should be 42 indeed (it's better to move it to a separate variable, smth like `const TOTAL_ITEMS = 42`).

Sorry for confusing